### PR TITLE
Assignment 6 branch

### PR DESCRIPTION
**Pull request by sebastienbeauvais**
Not really working. docker-compose up should run the Dockerfile and make a 'baseball' database, but it is not getting populated at this time.

### CODE REVIEW

**Comment by sebastienbeauvais** [comment]
**NOTES ON THIS ASSIGNMENT**
There are several files used here. The assignment_6.sql files makes the tables that the other SQL files will write to output files for. 100_day_rolling.sql create the 100 day rolling average table for all players. It has been modified since assignment 2. The 12560_rolling.sql is just for game 12560.

I am not sure if there are actual output files that docker gets from the bashfile.sh, so the assignment_6.sql can also output a txt file if you uncomment the last two lines.

For some reason even when specifying the version of MariaDB and upgrading pip the program would not take it and revert back to an older version (that actually is not on my machine). For this reason the wget command and the pip install --upgrade pip command remained in the Dockerfile. My guess is you will need to comment this out to run on your system. I tried running these commands in the terminal and still it would throw the error. 

weird bug:
- sometimes it will decide a table exists even if I have a fresh build. To fix this I would manually write DROP table [table_name] and re-build/up and it would solve?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     python3-dev \
+     mariadb-client \
+  && rm -rf /var/lib/apt/lists/*
+
+# Get necessary python libraries
+COPY requirements.txt .
+RUN wget https://dlm.mariadb.com/2678574/Connectors/c/connector-c-3.3.3/mariadb-connector-c-3.3.3-debian-bullseye-amd64.tar.gz -O - | tar -zxf - --strip-components=1 -C /usr
+RUN pip install --upgrade pip
+#RUN pip install mariadb
+RUN pip3 install --compile --no-cache-dir -r requirements.txt
```
Comment (cidb123) [inline Dockerfile:22]: Takes forever to run requirements, same with my code, maybe requirements not needed in the final or this assignment could be withheld?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
+FROM --platform=linux/amd64 python:3.10.6-buster
+
```
Comment (cidb123) [inline Dockerfile:2]: I get a strange warning here the same with my code, "The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested" maybe just switching it to --platform=linux/amd64/v8 could resolve future conflicts

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ENV APP_HOME /app
+WORKDIR $APP_HOME
+ENV PYTHONPATH /
+
```
Comment (cidb123) [inline Dockerfile:6]: Also docker-compose up is saying it's finding orphan containers which I think could just be from my docker having a project with the same name.

File: bashfile.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
```
Comment (cidb123) [inline bashfile.sh:2]: I'm seeing bashfile.sh permission denied currently but I fixed this by running chmod u+x bashscript.sh in terminal

File: homework/assignment_6/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+     python3 \
+     python3-pip \
+     python3-dev \
+     mariadb-client \
+  && rm -rf /var/lib/apt/lists/*
+
+# Get necessary python libraries
+COPY requirements.txt .
+RUN pip3 install --compile --no-cache-dir -r requirements.txt
+
```
Comment (cidb123) [inline homework/assignment_6/Dockerfile:20]: maybe add run chmod u+x samplescript.sh here?

Comment (cidb123) [reply homework/assignment_6/Dockerfile:20]: or maybe its a local error

File: homework/assignment_6/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+# Copy over code
+COPY flask_examples flask_examples
+COPY models models
+COPY my_super_sqlcode.sql
+COPY my_awesome_bash_script.sh
+
+# Run app
+CMD gunicorn --bind :$PORT --timeout 500 --workers 1 --threads 2 flask_examples.simple_pickled_model:app
```
Comment (cidb123) [inline homework/assignment_6/Dockerfile:28]: not sure we need this line, mine was running without it

File: homework/assignment_6/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    volumes:
+      - ./.db:/config
+    ports:
+      - 5306:3306
+    restart: unless-stopped
+
+  #baseball:
+  #      image: ubuntu
+  #      restart: unless-stopped
+  #      volumes:
```
Comment (cidb123) [inline homework/assignment_6/docker-compose.yml:22]: I don't think wee need these comments

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    environment:
+      - PUID=501
+      - PGID=20
+      - MYSQL_ROOT_PASSWORD=password
+      - TZ=Europe/London
+    volumes:
+      - ./.db:/config
+    ports:
+      - "5306:3306"
+    restart: always
```
Comment (cidb123) [inline docker-compose.yml:16]: Once code stopped both containers were still running, you may want to have these as "no" if you don't need them anymore

**Review by cidb123 — COMMENTED** [review]
Code runs well after I fixed the permissions issue, however I'm not sure where to find your output files.

**Pull request merged**